exporter/humioexporter/factory_test.go

```diff
@@ -118,13 +120,20 @@ func TestCreateTracesExporter(t *testing.T) {
 				tC.cfg,
 			)
 
-			if (err != nil) != tC.wantErr {
-				t.Errorf("CreateTracesExporter() error = %v, wantErr %v", err, tC.wantErr)
+			if (err != nil) != tC.wantErrorOnCreate {
+				t.Errorf("CreateTracesExporter() error = %v, wantErr %v", err, tC.wantErrorOnCreate)
 			}
 
 			if (err == nil) && (exp == nil) {
 				t.Error("No trace exporter created despite no errors")
 			}
+
+			if exp != nil {
+				err = exp.Start(context.Background(), componenttest.NewNopHost())
+				if (err == nil) && tC.wantErrorOnStart {
```

Comment:
This does not seem to handle the case where `err != nil` and we did not expect an error on the call to `Start`. I think it would be better to adopt the notation used on line 123:
```suggestion
				if (err != nil) != tC.wantErrorOnStart {
```